src/catalog/src/builtin.rs

```diff
@@ -5543,20 +5573,39 @@ SELECT
     inherit AS rolinherit,
     mz_catalog.has_system_privilege(r.oid, 'CREATEROLE') AS rolcreaterole,
     mz_catalog.has_system_privilege(r.oid, 'CREATEDB') AS rolcreatedb,
-    -- We determine login status each time a role logs in, so there's no way to accurately depict
-    -- this in the catalog. Instead we just hardcode NULL.
-    NULL::pg_catalog.bool AS rolcanlogin,
+    -- We determine login status each time a role logs in, so there's no clean
+    -- way to accurately determine this in the catalog. Instead we do something
+    -- a little gross. For system roles, we hardcode the known roles that can
+    -- log in. For user roles, we determine `rolcanlogin` based on whether the
+    -- role name looks like an email address.
+    --
+    -- This works for the vast majority of cases in production. Roles that users
+    -- log in to come from Frontegg and therefore *must* be valid email
+    -- addresses, while roles that are created via `CREATE ROLE` (e.g.,
+    -- `admin`, `prod_app`) almost certainly are not named to look like email
+    -- addresses.
+    --
+    -- For the moment, we're comfortable with the edge cases here. If we discover
+    -- that folks are regularly creating non-login roles with names that look
+    -- like an email address (e.g., `admins@sysops.foocorp`), we can change
+    -- course.
+    (
+        r.name IN ('mz_support', 'mz_system')
+        -- This entire scheme is sloppy, so we intentionally use a simple
+        -- regex to match email addresses, rather than one that perfectly
+        -- matches the RFC on what constitutes a valid email address.
+        OR r.name ~ '^[^@]+@[^@]+\.[^@]+$'
+    ) AS rolcanlogin,
```

Comment:
@MaterializeInc/adapter any qualms about this? Seemed like the best of the available options. We can always figure out how to do something more robust in the future.

Reply:
I don't love it, but I also don't have any better suggestions.